dotCMS/src/integration-test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java

```diff
@@ -457,499 +401,249 @@ public void checkInvalidActionTest()
-            vanityURLCached = vanityUrlCache.get(
-                    new CacheVanityKey(
-                            vanityURL1.getStringProperty(VanityUrlContentType.SITE_FIELD_VAR),
-                            vanityURL1.getLanguageId(),
-                            vanityURL1.getStringProperty(VanityUrlContentType.URI_FIELD_VAR)
-                    ));
-            Assert.assertNull(vanityURLCached);
+        // we have a spanish vanity because it is published
+        assert(spanishVanity.isPresent());
+        assertEquals(spanishVanity.get().vanityUrlId, englishVanity.get().vanityUrlId);
+        assertEquals(spanishVanity.get().languageId,spanish.getId());
+        
+        // unpublish the english one
+        filtersUtil.unpublishVanityURL(vanityURLContentletEnglish);
+        
+        
+        englishVanity = vanityUrlAPI.resolveVanityUrl("/testing" + currentTime, defaultHost, defaultLanguage);
+        assert(!englishVanity.isPresent());
 
-            //Request a vanity with a URL with a match
-            vanityURLCached = vanityUrlAPI
-                    .getLiveCachedVanityUrl(requestedURL, defaultHost, defaultLanguageId, user);
-            Assert.assertNotNull(vanityURLCached);
-            Assert.assertEquals(VanityUrlAPI.CACHE_404_VANITY_URL,
-                    vanityURLCached.getVanityUrlId());
+        
+        spanishVanity = vanityUrlAPI.resolveVanityUrl("/testing" + currentTime, defaultHost, spanish);
+        assert(spanishVanity.isPresent());
 
-            //------------------------------------
-            //Unpublish the VanityURL
-            //------------------------------------
-            vanityURL2.setIndexPolicy(IndexPolicy.FORCE);
-            vanityURL2.setIndexPolicyDependencies(IndexPolicy.FORCE);
-            filtersUtil.publishVanityUrl(vanityURL2);
-
-            //Should NOT be in cache
-            vanityURLCached = vanityUrlCache.get(
-                    new CacheVanityKey(
-                            vanityURL2.getStringProperty(VanityUrlContentType.SITE_FIELD_VAR),
-                            vanityURL2.getLanguageId(),
-                            vanityURL2.getStringProperty(VanityUrlContentType.URI_FIELD_VAR)
-                    ));
-            Assert.assertNull(vanityURLCached);
 
-            //Request a vanity with a URL with a match
-            vanityURLCached = vanityUrlAPI
-                    .getLiveCachedVanityUrl(requestedURL, defaultHost, defaultLanguageId, user);
-            Assert.assertNotNull(vanityURLCached);
-            Assert.assertNotEquals(VanityUrlAPI.CACHE_404_VANITY_URL,
-                    vanityURLCached.getVanityUrlId());
+    }
 
+    
 
-        } finally {
+    /**
+     * Testing how the cache is working with multiple vanities using the same URI and the
+     * combinations of existing vanities in other different Sites
+     */
+    @Test
+    public void does_a_site_specific_vanity_override_system_host_vanity()
+            throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)